provider/datagen/src/driver.rs

```diff
@@ -468,97 +468,160 @@ impl DatagenDriver {
     }
 }
 
+struct ExplicitImplicitLocaleSets {
+    explicit: HashSet<DataLocale>,
+    implicit: HashSet<DataLocale>,
+}
+
+/// Resolves the set of explicit langids and the supported locales into two sets of locales:
+///
+/// - `explicit` contains the explicit langids but with aux keys and extension keywords included
+/// - `implcit` contains any locale reachable by fallback from an `explicit` locale
```

Comment:
nit: I'd like more comments here to explain what this handles, with examples referencing the locales mentioned in comments below

Reply:
Added more docs